modules/scramble/clusteridentifier/main.nf

```diff
@@ -0,0 +1,40 @@
+process SCRAMBLE_CLUSTERIDENTIFIER {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::scramble=1.0.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/scramble:1.0.1--h779adbc_1':
+        'quay.io/biocontainers/scramble:1.0.1--h779adbc_1' }"
+
+    input:
+    tuple val(meta), path(input), path(input_index)
+    path fasta
+
+    output:
+    tuple val(meta), path("*.clusters.txt") , emit: clusters
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.0.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+
+    def reference = fasta ? "wget https://raw.githubusercontent.com/samtools/samtools/master/misc/seq_cache_populate.pl && perl seq_cache_populate.pl -root ./md5_ref ${fasta} && export REF_PATH=`pwd`/md5_ref/%2s/%2s/%s" : ""
```

Comment:
Can you comment here the second part of the reference?

Reply:
Can you explain why you do all that:
```
wget https://raw.githubusercontent.com/samtools/samtools/master/misc/seq_cache_populate.pl && perl seq_cache_populate.pl -root ./md5_ref ${fasta} && export REF_PATH=`pwd`/md5_ref/%2s/%2s/%s
```

Reply:
The tool does not contain a way to specify the reference file when using CRAM files. It just looks in the header of the CRAM file where the reference file is located, but that reference can't always be fetched since most test data is created on another machine. I had to find another way to specify the reference and I found that I could create an md5 cache of a specified fasta and supply it to the REF_PATH environment variable. This way the tool uses the correct reference. 

I've made an issue about this on the tool's repository to add a reference file argument, so this is just placeholder code until they added the argument.

Reply:
Can you add that information in a comment as well, and maybe a link to the issue just in case?

Reply:
Done :)